### PR TITLE
Add MapDatasetEventSampler sample_psf method

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR aims at adding a `sample_psf` method in the `MapDatasetEventSampler` class. The method applies PSF correction to the coordinates of the simulated events.

### CODE REVIEW

File: gammapy/cube/simulate.py [hunk 1]
```diff
@@ -176,3 +176,37 @@ def sample_background(self, dataset):
...
+        events : `EventList`
+            Event list.
+
+        Returns
+        -------
+        events : `EventList`
+            Event list with reconstructed position columns.
+        """
+
+        energy_unit = events.table["RA_TRUE"].unit
```
Comment (cdeil) [inline gammapy/cube/simulate.py]: This is an angular unit, not an energy unit, no?
Suggest to remove that line, and just will "deg" below.
That's what the spec requires: https://gamma-astro-data-formats.readthedocs.io/en/latest/events/events.html

File: gammapy/cube/simulate.py [hunk 2]
```diff
@@ -176,3 +176,37 @@ def sample_background(self, dataset):
...
+            PSF map.
+        events : `EventList`
+            Event list.
+
+        Returns
+        -------
+        events : `EventList`
+            Event list with reconstructed position columns.
+        """
+
```
Comment (cdeil) [inline gammapy/cube/simulate.py]: Remove empty line.

File: gammapy/cube/simulate.py [hunk 3]
```diff
@@ -176,3 +176,37 @@ def sample_background(self, dataset):
         table.rename_column("DEC_TRUE", "DEC")
 
         return EventList(table)
+
+    def sample_psf(self, psf_map, events):
+        """Sample psf map.
+
+        Parameters
+        ----------
+        psf_map : `PSFMap`
```
Comment (cdeil) [inline gammapy/cube/simulate.py]: You need to give full links to resolve, in this case:
```
~gammapy.cube.PSFMap
```
and below
```
~gammapy.data.EventList
```

File: gammapy/cube/tests/test_simulate.py [hunk 4]
```diff
@@ -116,3 +117,19 @@ def test_MDE_sample_background():
...
     assert_allclose(bkg_evt.table["MC_ID"][0], 0, rtol=1e-5)
+
+
+@requires_data()
+def test_MDE_sample_psf():
+    psf_map = make_test_psfmap(0.1 * u.deg, shape="gauss")
+
+    dataset = dataset_maker()
+    sampler = MapDatasetEventSampler(random_state=0)
+    src_evt = sampler.sample_sources(dataset=dataset)
```
Comment (cdeil) [inline gammapy/cube/tests/test_simulate.py]: Suggest to call `src_evt` and `src_evt_corr` both `events, like this:
```
events = sampler.sample_sources(dataset=dataset)
events = sampler.sample_psf(psf_map, events)
```
and then below you can have the asserts on `events.table`.

Comment (fabiopintore) [reply gammapy/cube/tests/test_simulate.py]: Done

File: gammapy/cube/simulate.py (hunk 3 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done

File: gammapy/cube/simulate.py (hunk 2 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done

File: gammapy/cube/simulate.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Yes, sorry. A mistake in bad copy and paste.
I've considered only 'deg'.

**Comment by adonath** [comment]
Thanks @fabiopintore!

**Pull request merged**